improve GETDEL description (#1508)
redis/redis-doc: commands/getdel.md

Get the value of `key` and delete the key.

> This command is similarly to `GET`, except for the fact that it also deletes the key on success (only if the key is string type).

Reviewer: ```suggestion
This command is similarl to `GET`, except for the fact that it also deletes the key on success (if and only if the key's value type is a string).
```

@return

> @bulk-string-reply: the value of `key`, `nil` when `key` does not exist, or error if the key is not a string type.

Reviewer: ```suggestion
@bulk-string-reply: the value of `key`, `nil` when `key` does not exist, or an error if the key's value type isn't a string.
```
